### ISSUE TITLE
Issue with display rows parameter

### ISSUE THREAD

**Issue by idomic** [ploomber/jupysql#43]
[This user](https://github.com/catherinedevlin/ipython-sql/issues/61) was having issues while using this parameter.
We need to reproduce and solve this.

**Comment by idomic** [comment]
cc: @szsiddiq

**Comment by edublancas** [comment]
I agree having `%config SqlMagic.displaylimit = 0` display all rows is odd, but I think we should keep backwards compatibility for now. Perhaps `%config SqlMagic.displaylimit = None` could hide all rows?

**Comment by idomic** [comment]
This looks like a bug, I don't think users are using it to show all rows?
We cna do a quick online search and see, but 0 doesn't look like a wanted functionality. If it is we can use none, it's a good workaround.

**Comment by edublancas** [comment]
Looks like this [has already been fixed](https://github.com/ploomber/jupysql/blob/0d7ecc82011a04e3198fd46ab6f90dec4778a345/src/sql/run.py#L420) since `0` is replaced by `None`. In any case, I think we should improve docs and add tests.

The other confusing thing is that there are two settings `autolimit` (automatically add a LIMIT to the queries) and `displaylimit` (limit rows displayed - even if more were fetched into memory from the db). And they have different default values (`0`, and `None`). I think we should add tests for both and pass `0` and `None`, then check that's the current behavior; then decide what behavior we want to be. If we decide to change defaults or existing behavior, we can add deprecation warnings.

This is what I think the most sensible behavior should be:

- autolimit (default: None)
     - None: should not any LIMIT clause to the queries
     - 0: either raise a warning (doesn't make sense for all queries to have `LIMIT 0`) or make it equal to `None`
- displaylimit (default: None)
     - None: display all rows
     - 0: do not display any rows

**Comment by idomic** [comment]
I think we can discuss this. looks like that line has an issue : `# TODO: remove this to make 0 really 0`
